core/domain/classifier_services.py

```diff
@@ -359,62 +416,56 @@ def delete_classifier_training_job(job_id):
         classifier_training_job_model.delete()
 
 
-def get_classifier_from_exploration_attributes(exp_id, exp_version,
-                                               state_name):
-    """Gets the classifier object from the exploration attributes.
+def get_classifier_training_jobs(exp_id, exp_version, state_names):
+    """Gets the classifier training job object from the exploration attributes.
 
     Args:
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version.
-        state_name: str. The name of the state to which the classifier
-            belongs.
+        state_names: list(str). The state names for which we retrieve the job.
 
     Returns:
-        ClassifierData. Domain object for the Classifier model.
+        list(ClassifierTrainingJob). Domain objects for the Classifier training
+            job model.
 
     Raises:
-        Exception: Entity for class ClassifierExplorationMapping with id not
+        Exception: Entity for class TrainingJobExplorationMapping with id not
```

Comment:
Perhaps, but remember you have the old list too. You can just iterate along it, adding an entry where you have a non-None original value, and adding a None otherwise.